packages/color-scale-block/src/components/SquareWithColor.tsx

```diff
@@ -0,0 +1,195 @@
+                .map((string: string) =>
+                    string.length === 1 ? "0" + string : string
+                ) // Adds 0 when length of one number is 1
+                .join("")
+        ); // Puts the array to together to a string
+    };
+
+    const copyToClipboard = async (text?: string) => {
+        try {
+            const toCopy = text || location.href;
+            await navigator.clipboard.writeText(toCopy);
+        } catch (err) {
+            console.error("Failed to copy: ", err);
+        }
+    };
+
+    return (
+        <div
+            style={{
+                height: height,
+                width: `${width}px`,
+                left: `${calculateLeftPosition(index, width)}px`,
+                display: "inline-block",
+            }}
+            id={`row-id-${id}-index-${index}`}
+            className={`hover:tw-z-30 row tw-overflow-visible tw-pb-8 tw-inline-block`}
+            key={id}
+        >
+            <DragHandle
+                index={index}
+                currentColor={currentColor}
+                isEditing={isEditing}
+                onResizeStart={onResizeStart}
+            />
+            <div
+                style={{
+                    height: height,
+                    borderTopLeftRadius: index === 0 ? "3px" : "0px",
+                    borderBottomLeftRadius: index === 0 ? "3px" : "0px",
+                    borderTopRightRadius:
+                        index === totalNumberOfBlocks - 1 ? "3px" : "0px",
+                    borderBottomRightRadius:
+                        index === totalNumberOfBlocks - 1 ? "3px" : "0px",
+                    borderLeftWidth: index === 0 ? "1px" : "0px",
+                    borderRightWidth:
+                        index === totalNumberOfBlocks - 1 ? "1px" : "0px",
+                    paddingTop: "1px",
+                    paddingBottom: "1px",
+                    paddingLeft: index === 0 ? "1px" : "0px",
+                    paddingRight:
+                        index === totalNumberOfBlocks - 1 ? "1px" : "0px",
+                    borderColor: "#efecec",
+                }}
+                ref={drag}
+                draggable={true}
+                onDrag={() => {
+                    if (isDragging !== currentColor.id) {
+                        setIsDragging(currentColor.id);
+                    }
+                }}
+                onDragEnd={() => {
+                    setIsDragging(false);
+                }}
+                className="tw-group tw-overflow-visible tw-top-2 tw-border tw-border-white tw-w-full hover:tw-border-black hover:tw-border"
+            >
+                <Tooltip
+                    alignment={TooltipAlignment.Middle}
+                    content={
+                        <div>
+                            <div>{currentColor.color.name}</div>
+                            <div>
+                                {rgbaToHex(
```

Comment:
Create const out of
`rgbaToHex(
                                    rgba(${currentColor.color.red}, ${currentColor.color.green}, ${currentColor.color.blue}, ${currentColor.color.alpha})
                                ).toUpperCase()`